evap/rewards/tools.py

```diff
@@ -106,22 +106,24 @@ def grant_reward_points_after_evaluate(request, semester, **_kwargs):
 def grant_reward_points_after_delete(instance, action, reverse, pk_set, **_kwargs):
     # if users do not need to evaluate a course anymore, they may have earned reward points
     if action == 'post_remove':
-        affected = []
+        granted = []
 
         if reverse:
             # a course got removed from a participant
             user = instance
 
             for semester in Semester.objects.filter(course__pk__in=pk_set):
-                if grant_reward_points_if_eligible(user, semester):
-                    affected = [user]
+                points_granted, __ = grant_reward_points_if_eligible(user, semester)
+                if points_granted:
+                    granted = [{"user": user, "points": points_granted}]
```

Comment:
now that's the prime example for a namedtuple :) my suggestion was probably unclear, i meant a single dictionary mapping from users to points, not a list of dicts. on second thought, i think the list of namedtuples would actually be okay, so you can choose, but a list of dictionaries kind of combines the disadvantages of both options :)

sorry for the back and forth.

...what about letting grant_reward_points_if_eligible return the actual granting object (or None, plus that boolean)? i mean, that contains the user and the points, so it's pretty much what we want here.